pkg/controllers/management/auth/legacy_rt_cleaner.go

```diff
@@ -2,23 +2,26 @@ package auth
 
 import (
 	"github.com/rancher/rancher/pkg/api/norman/store/roletemplate"
-	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	v3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	normanv3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
 	"github.com/rancher/rancher/pkg/types/config"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
 type rtCleaner struct {
-	mgmt *config.ManagementContext
+	roleTemplates v3.RoleTemplateController
+	clusterLister v3.ClusterController
 }
 
 func newLegacyRTCleaner(mgmt *config.ManagementContext) *rtCleaner {
 	return &rtCleaner{
-		mgmt: mgmt,
+		roleTemplates: mgmt.Wrangler.Mgmt.RoleTemplate(),
+		clusterLister: mgmt.Wrangler.Mgmt.Cluster(),
 	}
 }
 
 // sync cleans up all roleTemplates to drop cluster-scoped lifecycle handler finalizers
-func (p *rtCleaner) sync(key string, obj *v3.RoleTemplate) (runtime.Object, error) {
+func (p *rtCleaner) sync(key string, obj *normanv3.RoleTemplate) (runtime.Object, error) {
```

Comment:
There should be a non-norman equivalent of RoleTemplate.  We should probably use that instead.